internal/metamorph/processor.go

```diff
@@ -390,13 +411,10 @@ func (p *Processor) StartSendStatusUpdate() {
 					CompetingTxs: msg.CompetingTxs,
 				}
 
-				// if we receive new update check if we have client connection waiting for status and send it
-				p.responseProcessor.UpdateStatus(msg.Hash, StatusAndError{
-					Hash:         msg.Hash,
-					Status:       msg.Status,
-					Err:          msg.Err,
-					CompetingTxs: msg.CompetingTxs,
-				})
+				// if tx is rejected, we don't expect any more status updates - remove from cache
+				if msg.Status == metamorph_api.Status_REJECTED {
```

Comment:
As stated above I think this can be 

```suggestion
if msg.Status == metamorph_api.Status_REJECTED || msg.Status == metamorph_api.Status_SEEN_ON_NETWORK {
```

Reply:
You're right, added that